### PR TITLE
Fix bugs in dEQP texturefiltering_3d_{formats|size}_* tests

### PR DESCRIPTION
**Pull request by qkmiao**
With this patch, all texturefiltering_3d_formats/size_* tests passed.

### CODE REVIEW

File: sdk/tests/deqp/framework/common/tcuTexture.js [hunk 1]
```diff
@@ -223,7 +223,7 @@ tcuTexture.sampleLevelArray2DCompare = function(levels, numLevels, sampler, ref,
         case tcuTexture.FilterMode.NEAREST_MIPMAP_LINEAR:
         case tcuTexture.FilterMode.LINEAR_MIPMAP_LINEAR: {
             var maxLevel = numLevels - 1;
-            var level0 = deMath.clamp(Math.ceil(lod), 0, maxLevel);
+            var level0 = deMath.clamp(Math.floor(lod), 0, maxLevel);
```
Comment (qkmiao) [inline sdk/tests/deqp/framework/common/tcuTexture.js]: C++ code:
https://android.googlesource.com/platform/external/deqp/+/master/framework/common/tcuTexture.cpp#1909

File: sdk/tests/deqp/framework/common/tcuTexture.js [hunk 2]
```diff
@@ -2282,40 +2282,8 @@ tcuTexture.TextureLevelPyramid.prototype.clearLevel = function(levelNdx) {
...
-
-            return deMath.add(deMath.scale(t0, 1 - f), deMath.scale(t1, f));
-        }
-
-        default:
-            throw new Error('Invalid filter mode:' + filterMode);
-    }
-    throw new Error('Unimplemented');
+    // z-offset in 2D textures is layer selector
+    return tcuTexture.sampleLevelArray2DOffset(levels, numLevels, sampler, [s, t], lod, [0, 0, depth]);
```
Comment (qkmiao) [inline sdk/tests/deqp/framework/common/tcuTexture.js]: https://android.googlesource.com/platform/external/deqp/+/master/framework/common/tcuTexture.cpp#1715

File: sdk/tests/deqp/framework/common/tcuTexture.js [hunk 3]
```diff
@@ -2329,39 +2297,7 @@ tcuTexture.sampleLevelArray2D = function(levels, numLevels, sampler, s, t, depth
...
-            var t1 = levels[level1].sample3D(sampler, levelFilter, s, t, r);
-
-            return deMath.add(deMath.scale(t0, 1 - f), deMath.scale(t1, f));
-        }
-
-        default:
-           throw new Error('Invalid filter mode:' + filterMode);
-    }
-    throw new Error('Unimplemented');
+    return tcuTexture.sampleLevelArray3DOffset(levels, numLevels, sampler, s, t, r, lod, [0, 0, 0]);
```
Comment (qkmiao) [inline sdk/tests/deqp/framework/common/tcuTexture.js]: https://android.googlesource.com/platform/external/deqp/+/master/framework/common/tcuTexture.cpp#1720

File: sdk/tests/deqp/modules/shared/glsTextureTestUtil.js [hunk 4]
```diff
@@ -1655,8 +1655,12 @@ glsTextureTestUtil.computeTextureLookupDiff3D = function(
             qpWatchDog_touch(watchDog);*/
 
         for (var px = 0; px < result.getWidth(); px++) {
+            /** @type {Array<number>} */
             var resPix = result.getPixel(px, py);
+            glsTextureTestUtil.deapplyScaleAndBias(resPix, sampleParams.colorScale, sampleParams.colorBias);
+            /** @type {Array<number>} */
             var refPix = reference.getPixel(px, py);
+            glsTextureTestUtil.deapplyScaleAndBias(refPix, sampleParams.colorScale, sampleParams.colorBias);
```
Comment (qkmiao) [inline sdk/tests/deqp/modules/shared/glsTextureTestUtil.js]: https://android.googlesource.com/platform/external/deqp/+/master/modules/glshared/glsTextureTestUtil.cpp#2215

**Comment by zhenyao** [comment]
Great!

**Pull request merged**